XenAdmin/ConsoleView/RdpClient.cs

```diff
@@ -78,26 +75,26 @@ internal RdpClient(ContainerControl parent, Size size, EventHandler resizeHandle
                 rdpClient9.OnRemoteDesktopSizeChange += rdpClient_OnRemoteDesktopSizeChange;
 
                 // CA-96135: Try adding rdpControl to parent.Controls list; this will throw exception when
-                // MsRdpClient8 control cannot be created (there is no appropriate version of dll present)
+                // MsRdpClient9 control cannot be created (there is no appropriate version of dll present)
                 parent.Controls.Add(rdpControl);
                 allowDisplayUpdate = true;
             }
-            catch
+            catch (Exception ex)
             {
                 if (parent.Controls.Contains(rdpControl))
                     parent.Controls.Remove(rdpControl);
+                rdpControl.Dispose();
+                rdpControl = null;
                 rdpClient9 = null;
-                rdpControl = rdpClient6 = new MsRdpClient6();
-                RDPConfigure(size);
-                parent.Controls.Add(rdpControl);
+                Log.Error("Adding rdpControl to parent.Controls list caused an exception.", ex);

```

Comment:
Can we move the Resize registration inside the try block please ; if something goes wrong and the catch block is hit, rdpControl will be null.